### DISCUSSION TITLE
submariner-gateway pod getting restarted repeatedly

### DISCUSSION THREAD

**Discussion by yashpratap10** [submariner-io/submariner#1940] [Q&A] [answered]
I deployed Submariner in my multi-server redundant layout where one is a primary server where submariner-broker is installed and the other is a redundant secondary server. I'm trying to install with subctl utility, and all the pods are getting started successfully except the submariner-gateway pod which is restarting repeatedly. I can see this below error in the pod logs - 

```
Error resolving public IP with resolver api:api.ipify.org : retrieving public IP from https://api.ipify.org: Get "https://api.ipify.org"
Error resolving public IP with resolver api:api.my-ip.io/ip : retrieving public IP from https://api.my-ip.io/ip: Get "https://api.my-ip.io/ip"
Error resolving public IP with resolver api:ip4.seeip.org : retrieving public IP from https://ip4.seeip.org: Get "https://ip4.seeip.org"
Error creating local endpoint object: Unable to resolve public IP by any of the resolver methods: [api:api.ipify.org api:api.my-ip.io/ip api:ip4.seeip.org]
```
Environment config: 
Rancher Kubernetes engine
Submariner version:v0.12.1
kubernetes version:v1.19.16

Please let me know how can I resolve this issue or any other config info needed to look into the issue more closely.

**Comment by sridhargaddam** [comment, marked as answer]
Couple of observations:

1) Is your broker cluster accessible from the secondary cluster? If not, its necessary that you install Broker on a cluster that is accessible from all the managed clusters. Submariner has a set of pre-requisites as listed here - https://submariner.io/getting-started/#prerequisites

2) From the logs you pasted, it looks like the submariner-gateway pod is unable to resolve the public-ip. Does the Gateway node has access to internet? or is it an air-gapped deployment which does not have access to internet?

3) Depending on your topology -  If the public-ip cannot be resolved, you can still manually annotate the Gateway nodes in the cluster with the public-ip as shown in the page https://submariner.io/operations/nat-traversal/
Also, if there is a double-NAT scenario, you can follow this - https://github.com/submariner-io/submariner-website/pull/806
